### DISCUSSION TITLE
How to call a cython library from pyo3?

### DISCUSSION THREAD

**Discussion by DennisZhangOiler** [PyO3/pyo3#3623] [Questions]
I have a shared library that is built by python codes, how can i call it from rust?

**Comment by davidhewitt** [comment]
I think you'd expect to interact with it through `PyModule::import` as if it were any other Python module; the fact that it's Cython should be treated as an implementation detail.

**Comment by DennisZhangOiler** [comment]
> I think you'd expect to interact with it through `PyModule::import` as if it were any other Python module; the fact that it's Cython should be treated as an implementation detail.

Thanks for replying, I may have some confusion here since I am not that familiar with python and cython, what I mean is I have a shared library built by command `python3 -m Cython.Build.Cythonize -3b *.py;rm *.c;rm -rf build;`, and in python it can be directly imported under the same path, but if i call importing in pyo3, it will report module not found, would you mind helping me with it?

**Reply by davidhewitt** [reply]
I think https://pyo3.rs/v0.20.0/python_from_rust is a good starting point maybe.

What os are you using? Possibly #1896 (and similar discussion for macos) may get in the way.

**Reply by DennisZhangOiler** [reply]
Hi I am using ubuntu, it failed to import even I manually linked into the bin, I still can not figure out how to do it.
